apps/validator/src/main/scala/org/lfdecentralizedtrust/splice/validator/store/ValidatorConfigProvider.scala

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package org.lfdecentralizedtrust.splice.validator.store
+
+import cats.data.OptionT
+import com.digitalasset.canton.tracing.TraceContext
+import io.circe.{Decoder, Encoder}
+import scala.concurrent.{ExecutionContext, Future}
+import io.circe.syntax.*
+
+final case class ScanUrlInternalConfig(
+    svName: String,
+    url: String,
+)
+
+class ValidatorConfigProvider(config: ValidatorInternalStore) {
+
+  private val scanInternalConfigKey = "validator_scan_internal_config_key"
+
+  implicit val scanUrlEncoder: Encoder[ScanUrlInternalConfig] =
+    Encoder.forProduct2("svName", "url")(s => (s.svName, s.url))
+  implicit val scanUrlDecoder: Decoder[ScanUrlInternalConfig] =
+    Decoder.forProduct2("svName", "url")(ScanUrlInternalConfig.apply)
```

Comment:
nit: Ideally we would have this in the companion object for `ValidatorConfigProvider`, and `ScanUrlInternalConfig` as well.
also we should use semi automatic derivation to make our life easier
```suggestion
  implicit val scanUrlCodec: Codec[ScanUrlInternalConfig] = deriveCodec[ScanUrlInternalConfig]
    
```